packages/apps/desktop-viewer-test/src/frontend/components/App.tsx

```diff
@@ -3,49 +3,61 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 
+import { NativeApp } from "@bentley/imodeljs-frontend";
 import { useDesktopViewerInitializer } from "@itwin/desktop-viewer-react";
 import { Router } from "@reach/router";
 import React, { useCallback, useEffect, useState } from "react";
 
 import { ViewerSettings } from "../../common/ViewerConfig";
+import { ITwinViewerApp } from "../app/ITwinViewerApp";
 import {
-  addRecentOnline as addRecentOnlineClient,
-  addRecentSnapshot as addRecentSnapshotClient,
+  addRecent as addRecentClient,
   getUserSettings,
   SettingsContext,
 } from "../services/SettingsClient";
-import {
-  DownloadRoute,
-  HomeRoute,
-  IModelsRoute,
-  ITwinsRoute,
-  SnapshotRoute,
-} from "./routes";
+import { HomeRoute, IModelsRoute, ITwinsRoute, ViewerRoute } from "./routes";
 
 const App = () => {
   const initialized = useDesktopViewerInitializer();
   const [settings, setSettings] = useState<ViewerSettings>();
+  const [connectivityListener, setConnectivityListener] =
+    useState<() => void>();
 
   useEffect(() => {
     if (initialized) {
+      // setup connectivity events to let the backend know the status
+      void NativeApp.checkInternetConnectivity().then((status) => {
+        void ITwinViewerApp.ipcCall.setConnectivity(status);
+        const listener = NativeApp.onInternetConnectivityChanged.addListener(
+          (status) => {
+            void ITwinViewerApp.ipcCall.setConnectivity(status);
+          }
+        );
+        setConnectivityListener(listener);
+      });
       void getUserSettings().then((userSettings) => {
         setSettings(userSettings);
       });
     }
+    return () => {
+      if (connectivityListener) {
+        connectivityListener();
+      }
+    };
   }, [initialized]);
```

Comment:
I don't think so. This is where we set connectivityListener, so it would cause an un-necessary effect....its only being referenced in the return statement (to basically remove the listener) under the assumption that it is not being set anywhere else.

Reply:
its saved so that it can be removed....otherwise it would be a potential memory leak